src/jit/importer.cpp

```diff
@@ -3388,6 +3389,20 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
     {
         // The recursive calls to Jit intrinsics are must-expand by convention.
         mustExpand = mustExpand || gtIsRecursiveCall(method);
```

Comment:
If we're now always expanding calls to HW intrinsics then isn't this comment and logic out of date?

Reply:
It might be.

I wasn't sure if there are other JIT intrinsics, which can be recursive, but for which we do not want to always expand.

Reply:
We added this bit just for HW intrinsics.

Reply:
Then should we rename the bit to indicate that it is exclusively for HWIntrinsics (CORINFO_FLG_HW_INTRINSIC) or that this bit will assume "mustExpand = true" (CORINFO_FLG_MUSTEXPAND_INTRINSIC).

Based on the current logic in the VM (https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L5144) this bit is set for any method marked with `[Intrinsic]`

Reply:
Ah, 'bit' was a poor choice of words. We added this bit of logic in the importer just for HW intrinsics.

Not all `[Intrinsic]` methods are must expand; some of them have perfectly viable IL implementations.

Reply:
Ok, I think I get it 😄 

Then my question is: Do we expect all intrinsics for which `gtIsRecursiveCall()` would return true to always expand (except for indirect invocation) or do we only expect them to always expand for hardware intrinsics?

Even in the case of the former, I'm not sure how we detect the difference between a JitIntrinsic that is recursive and one that has an IL implementation on first pass..